### PR TITLE
Add prototype MySQL mixin.

### PR DESCRIPTION
**Pull request by tomwilkie**
Based off of the dashboard at https://grafana.com/grafana/dashboards/11323, but with:
- Variables for datasource, job and instance
- Use of $__instance metavar for rate
- sum aggregations to reduce cardinality of dashboards
- removed node-exporter panels.

TODO:
- [x] CI for build & lint
- [x] Alerts
- [x] Audit dashboard for meaningful queries.

Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

### CODE REVIEW

**Comment by bergquist** [comment]
ping @SuperQ

File: mysqld-mixin/go.mod [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module github.com/prometheus/mysqld_exporter/mysqld_mixin
+
+go 1.14
+
+require (
+	github.com/google/go-jsonnet v0.16.0 // indirect
+	github.com/monitoring-mixins/mixtool v0.0.0-20201004122653-ee7999927a3a // indirect
+)
```
Comment (sh0rez) [inline mysqld-mixin/go.mod:8]: Is this expected? Seems like an artifact of `go install`

**Review by sh0rez — APPROVED** [review]
Looking good! 

If we are going to adopt the YAML strategy for mixins, does this mean we expect them to be static? ie. not take config parameters (selectors, etc)?

File: mysqld-mixin/go.mod (hunk 1 above)
Comment (tomwilkie) [reply mysqld-mixin/go.mod:8]: This is there to pin the version of the mixtool and jsonnetfmt we use in CI, so it doesn't break when they change :-)

(This pattern is what we do in node mixin too)

File: mysqld-mixin/Makefile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+JSONNET_FMT := jsonnetfmt -n 2 --max-blank-lines 2 --string-style s --comment-style s
+
+all: fmt lint build clean
```
Comment (kavirajk) [inline mysqld-mixin/Makefile:3]: better to add `default: build`. 

Currently it defaults to `all`.  (meaning it builds and removes the dashboards as well)
idea is to run `build` step when I just type `make` :) more intuitive!

File: mysqld-mixin/.gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+alerts.yaml
```
Comment (bergquist) [inline mysqld-mixin/.gitignore:1]:
```suggestion
/alerts.yaml
```

File: mysqld-mixin/.gitignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+rules.yaml
```
Comment (bergquist) [inline mysqld-mixin/.gitignore:2]:
```suggestion
/rules.yaml
```

File: mysqld-mixin/go.mod [hunk 5]
```diff
@@ -0,0 +1,9 @@
+module github.com/prometheus/mysqld_exporter/mysqld_mixin
```
Comment (sdboyer) [inline mysqld-mixin/go.mod:1]:
```suggestion
module github.com/prometheus/mysqld_exporter/mysqld-mixin
```

File: mysqld-mixin/alerts/general.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+groups:
+- name: MySQLdAlerts
+  rules:
+  - alert: MySQLUp
```
Comment (jdbaldry) [inline mysqld-mixin/alerts/general.yaml:4]:
```suggestion
  - alert: MySQLDown
```

File: mysqld-mixin/README.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
+# MySQLd Mixin
+
+_This is a work in progress. We aim for it to become a good role model for alerts
+and dashboards eventually, but it is not quite there yet._
```
Comment (SuperQ) [inline mysqld-mixin/README.md:4]: Let's drop this, it smells of the :construction: banner. :smile_cat:

File: mysqld-mixin/rules/rules.yaml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+groups:
+- name: mysqld_rules
+  rules:
+
+  # Record slave lag seconds for pre-computed timeseries that takes
+  # `mysql_slave_status_sql_delay` into account
+  - record: :mysql_slave_lag_seconds
```
Comment (SuperQ) [inline mysqld-mixin/rules/rules.yaml:7]: I'm not a fan of making metric names start with a bare `:`. Is this a mixin pattern? Typically if I am not using an aggregation, I start rules like this with `instance`. So, `instance:mysql_slave_lag_seconds`.

**Review by SuperQ — COMMENTED** [review]
A couple of minor nits, otherwise seems good.

File: mysqld-mixin/rules/rules.yaml (hunk 8 above)
Comment (tomwilkie) [reply mysqld-mixin/rules/rules.yaml:7]: I'm not a fan either - the rule I was applying was to make sure all recording rules have a colon in.  I like your suggestion though.

File: mysqld-mixin/README.md (hunk 7 above)
Comment (tomwilkie) [reply mysqld-mixin/README.md]: :+1:

**Comment by tomwilkie** [comment]
Thanks for the feedback Ben! 

Let me know when you're happy to merge.

**Review by SuperQ — APPROVED** [review]
LGTM

**Pull request merged**